src/OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed/NativeMethods.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace OpenTelemetry.AutoInstrumentation.ClrProfiler.Managed
+{
+    internal static class NativeMethods
+    {
+        private static readonly bool IsWindows = string.Equals(FrameworkDescription.Instance.OSPlatform, "Windows", StringComparison.OrdinalIgnoreCase);
+
+        public static bool IsProfilerAttached()
+        {
+            if (IsWindows)
+            {
+                return Windows.IsProfilerAttached();
+            }
+
+            return NonWindows.IsProfilerAttached();
+        }
+
+        // the "dll" extension is required on .NET Framework
+        // and optional on .NET Core
+        private static class Windows
+        {
+            [DllImport("Datadog.Trace.ClrProfiler.Native.dll")]
```

Comment:
This dll was renamed.

Reply:
Perfect catch... I was looking why the hell `IsProfilerAttached` logging `false` all the time 😄

Reply:
Seems I have to do fix PR for native lib naming before for Linux / MacOS. No point to do half change (fix for Windows only) IMO.